### PR TITLE
Add Best Practices Section on Subscriptions

### PR DESCRIPTION
**Pull request by oldbonsai**
Changes proposed in this pull request:

* Add a new section for best practices that outlines how to respect the restrictions that subscriptions put on content.

---

The credentials to view the Netlify deploy is the following:

* **username**: crabigator
* **password**: noblowfishallowed

### CODE REVIEW

File: source/includes/20170710/_best_practices.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
 # Best Practices
 
-We're always working to make the API as performant as possible, but there are a few things you can do to optimize your use of the data we deliver and speed things up when you need to make new requests.
+We're always working to make the API as performant as possible, but there are a few things you can do to optimize your use of the data we deliver and speed things up when you need to make new requests: [cache data locally](#caching) whenever possible, [make conditional requests](#conditional-requests) to minimize network load, and [make use of the `updated_after` filter](#leveraging-the-code-updated_after-code-filter) on a lot of the endpoints.
+
+When you're building applications or services that other people will use, there's also some work to be done to respect the access to content granted by a subscription to WaniKani (per our terms and generally being a good citizen).
```
Comment (vietqhoang) [inline source/includes/20170710/_best_practices.md]: Should we anchor link to the subscription section from here?

File: source/includes/20170710/best_practices/_respecting_subscription_restrictions.md.erb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+The `user` endpoint has a `subscription` attribute. That should have everything you need.
+
+* `max_level_granted` defines the maximum level of content that's available to the user. It has two possible values: 3 and 60. The user shouldn't be able to access subjects above those levels. Lessons and reviews above those levels shouldn't be available at all and will be rejected if you try and submit them to us.
+* `active` is a boolean that tells you if the person has an active subscription.
+* `type` defines the kind of subscription, and works closely with `period_ends_at`. There are three values:
+  * `free` subscriptions aren't really subscriptions, but can represent people who've never subscribed or have an inactive subscription. There's no `period_ends_at` for free subscriptions.
+  * `recurring` subscriptions renew on a periodic basis. `period_ends_at` tells you when the subscription renews. Since we give people access until the end of their subscription period even if they cancel, you can safely not check their subscription status until that time. We put this in to help with both caching and offline access.
```
Comment (vietqhoang) [inline source/includes/20170710/best_practices/_respecting_subscription_restrictions.md.erb]: `period_ends_at` isn't always the case when the subscription renews. It is also the end of the user's subscription if they choose to not auto renew. Maybe a rephrasing is best here.

File: source/includes/20170710/best_practices/_respecting_subscription_restrictions.md.erb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  * `lifetime` means the user can access WaniKani forever. `period_ends_at` is null, mainly because `∞` is hard for computers to get. You don't need to schedule checks on the subscription status for this person — we don't take away lifetime subscriptions.
```
Comment (vietqhoang) [inline source/includes/20170710/best_practices/_respecting_subscription_restrictions.md.erb]: I would rephrase `You don't need to schedule checks on the subscription status for this person`.

There are cases when a lifetime subscription is removed. One example is if someone asks for a refund or if there is a charge back.

Thus a system is required to check this status.

**Comment by oldbonsai** [comment]
@vietqhoang Ready for a re-review when you have some time.
